### PR TITLE
feat: spawn components based on roles

### PR DESCRIPTION
**Pull request by hadjiszs**
# Description

- Spawning the different processes based on the config roles
- Adapt the graceful shutdown and the related tests
- Adapt some tests around the config file format
- Brought one simple keystore (`topos-wallet` that differs from `topos-crypto`) to ease the key sharing between TCE and Edge
   - For now rely on the local fs, but eventually be extended to handle the AWS SM same as in the Edge side

Fixes TP-630

## PR Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added or updated tests that comprehensively prove my change is effective or that my feature works

### CODE REVIEW

File: crates/topos/src/components/tce/mod.rs [hunk 1]
```diff
@@ -170,7 +177,7 @@ pub(crate) async fn handle_command(
 }
 
 pub fn print_node_info(config: &TceConfiguration) {
-    tracing::warn!("TCE Node - version: {}", config.version);
+    tracing::warn!("Topos - version: {}", config.version);
```
Comment (atanmarko) [inline crates/topos/src/components/tce/mod.rs:180]: This should be `info` probably not trace

File: crates/topos/src/edge/mod.rs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                        Err(_) => println!("{l}"),
+                    },
+                    None => break,
+                }
+            }
+        };
+
+        let (running_out, _) = tokio::join!(running, logging);
+
+        debug!("The Edge process is terminated");
```
Comment (atanmarko) [inline crates/topos/src/edge/mod.rs:88]: `info`

**Review by atanmarko — APPROVED** [review]

**Review by gruberb — APPROVED** [review]

File: crates/topos/src/components/node/mod.rs [hunk 3]
```diff
@@ -1,40 +1,71 @@
...
 pub(crate) async fn handle_command(
     NodeCommand {
         subcommands,
-        verbose: _,
+        verbose,
         home,
+        edge_path,
     }: NodeCommand,
 ) -> Result<(), Box<dyn std::error::Error>> {
+    setup_tracing(verbose, None, None)?;
```
[the remark this answers is not in the document]
Comment (hadjiszs) [reply crates/topos/src/components/node/mod.rs:67]: ah in the `Up` flow I am not forwarding the command `TceCommands::Run`, I'm calling directly the `topos_tce::run` if the TCE component has to be started

File: crates/topos/src/components/node/mod.rs [hunk 4]
```diff
@@ -1,40 +1,71 @@
...
 };
+use services::*;
+use tokio::process::Command;
+use topos_tce::config::{StorageConfiguration, TceConfiguration};
+use topos_wallet::SecretManager;
 
 pub(crate) mod commands;
+pub mod services;
+
+pub fn generate_edge_config(
```
[the remark this answers is not in the document]
Comment (hadjiszs) [reply crates/topos/src/components/node/mod.rs:47]: I moved it to `components/node/services.rs` instead, aside the other triggers for `topos node`

**Pull request merged**